src/main/java/org/osiam/client/OsiamConnector.java

```diff
@@ -129,48 +124,131 @@ public static void setMaxConnections(int maxConnections) {
+     * to construct one.
      *
-     * @param builder a valid Builder that holds all needed variables
-     * @throws IllegalStateException If OSIAM's endpoint is not configured
+     * @param builder a valid {@link Builder} that holds all needed variables
      */
     private OsiamConnector(Builder builder) {
-        checkState(!Strings.isNullOrEmpty(builder.endpoint), "OSIAM's endpoint is not configured.");
+        String authEndpoint;
+        String resourceEndpoint;
+        if (!Strings.isNullOrEmpty(builder.endpoint)) {
+            authEndpoint = builder.endpoint;
+            resourceEndpoint = builder.endpoint;
+        } else {
+            authEndpoint = authServiceEndpoint(builder);
+            resourceEndpoint = resourceServiceEndpoint(builder);
+        }
 
-        String endpoint = cleanEndpoint(builder.endpoint);
-        AuthService.Builder authServiceBuilder = new AuthService.Builder(endpoint);
-        if (builder.clientId != null) {
-            authServiceBuilder = authServiceBuilder.setClientId(builder.clientId);
+        if (!Strings.isNullOrEmpty(authEndpoint)) {
+            AuthService.Builder authServiceBuilder = new AuthService.Builder(authEndpoint);
+            if (builder.clientId != null) {
+                authServiceBuilder = authServiceBuilder.setClientId(builder.clientId);
+            }
+            if (builder.clientSecret != null) {
+                authServiceBuilder = authServiceBuilder.setClientSecret(builder.clientSecret);
+            }
+            if (builder.clientRedirectUri != null) {
+                authServiceBuilder = authServiceBuilder.setClientRedirectUri(builder.clientRedirectUri);
+
+            }
+            authService = authServiceBuilder
+                    .withConnectTimeout(builder.connectTimeout)
+                    .withReadTimeout(builder.readTimeout)
+                    .build();
         }
-        if (builder.clientSecret != null) {
-            authServiceBuilder = authServiceBuilder.setClientSecret(builder.clientSecret);
+
+        if (!Strings.isNullOrEmpty(resourceEndpoint)) {
+            userService = new OsiamUserService.Builder(resourceEndpoint)
+                    .withConnectTimeout(builder.connectTimeout)
+                    .withReadTimeout(builder.readTimeout)
+                    .withLegacySchemas(builder.legacySchemas)
+                    .build();
+            groupService = new OsiamGroupService.Builder(resourceEndpoint)
+                    .withConnectTimeout(builder.connectTimeout)
+                    .withReadTimeout(builder.readTimeout)
+                    .withLegacySchemas(builder.legacySchemas)
+                    .build();
         }
-        if (builder.clientRedirectUri != null) {
-            authServiceBuilder = authServiceBuilder.setClientRedirectUri(builder.clientRedirectUri);
+    }
+
+    private String authServiceEndpoint(Builder builder) {

```

Comment:
I'm not very happy with the names chosen for this methods. In my opinion they deserve a verb in the method name. I can see what you are getting at here but I think we did it nowhere else yet.